perfkitbenchmarker/pkb.py

```diff
@@ -198,69 +165,76 @@ def ListUnknownBenchmarks():
                 valid_benchmark_sets)
 
 
-def RunBenchmark(benchmark, collector):
+def RunBenchmark(benchmark, collector, sequence_number, total_benchmarks):
   """Runs a single benchmark and adds the results to the collector.
 
   Args:
     benchmark: The benchmark module to be run.
     collector: The SampleCollector object to add samples to.
+    sequence_number: The sequence number of when the benchmark was started
+      relative to the other benchmarks in the suite.
+    total_benchmarks: The total number of benchmarks in the suite.
   """
   benchmark_info = benchmark.GetInfo()
   benchmark_specification = None
   if not ValidateBenchmarkInfo(benchmark_info):
     return
-
-  # Optional prerequisite checking.
-  check_prereqs = getattr(benchmark, 'CheckPrerequisites', None)
-  if check_prereqs:
+  benchmark_name = benchmark_info['name']
+
+  # Modify the logger prompt for messages logged within this function.
+  label_extension = '{}({}/{})'.format(
+      benchmark_name, sequence_number, total_benchmarks)
+  log_context = log_util.GetThreadLogContext()
+  with log_util.ThreadLogContext.LabelExtension(log_context, label_extension):
```

Comment:
Prefer to call the instance method rather than passing to instance to the class:
```{python}
with log_context.LabelExtension(label_extension):
  ...
```